src/blob/file.rs

```diff
@@ -4,130 +4,59 @@ const WOULDBLOCK_RETRY_INTERVAL_MS: u64 = 10;
 
 #[derive(Debug, Clone)]
 pub(crate) struct File {
-    pub(crate) read_fd: Arc<fs::File>,
-    pub(crate) write_fd: Arc<Mutex<fs::File>>,
+    read_fd: Arc<StdFile>, // requires only for read_at/write_at methods
```

Comment:
`read_fd` makes impression that it used for read-only operations and `write_fd` for write-only. May be better to name it something like this: `no_lock_fd` and `write_fd` to something like `cursor_fd`

Reply:
fixed in #49 PR